templates/web/pages/tag/tag.tt.html

```diff
@@ -56,10 +63,36 @@
     </p>
 [% END %]
 
+
 [% IF products_title %]
     <h2 class="products_title">[% products_title %]</h2>
 [% END %]
         </div>
     </div>
 </div>
-<!-- end templates/[% template.name %] -->
\ No newline at end of file
+    
+</div>
+
+<!-- Fetching facet knowledge panel -->
+<script>
+    let fecet_kp_url = "https://facets-kp.openfoodfacts.net/render-to-html?"
+    let param = "facet_tag=[% tagtype_name %]&value_tag=[% tag_name %]&sec_facet_tag=[% tagtype2_name %]&sec_value_tag=[% tag2_name %]"
```

Comment:
what parameters do you need to pass to facets-kp?

today the template is passed those values:

			# Pass template data to generate navigation links
			$tag_template_data_ref->{tagtype} = $tagtype;
			$tag_template_data_ref->{tagtype_path} = '/' . $tag_type_plural{$tagtype}{$lc};
			$tag_template_data_ref->{tagtype_name} = lang($tagtype . '_s');
			$tag_template_data_ref->{tagid} = $tagid;
			$tag_template_data_ref->{tagid_path} = $newtagidpath;
			$tag_template_data_ref->{tag_name} = $display_tag;

			if (defined $tagid2) {
				$tag_template_data_ref->{tagtype2} = $tagtype2;
				$tag_template_data_ref->{tagtype2_path} = '/' . $tag_type_plural{$tagtype2}{$lc};
				$tag_template_data_ref->{tagtype2_name} = lang($tagtype2 . '_s');
				$tag_template_data_ref->{tagid2} = $tagid2;
				$tag_template_data_ref->{tagid2_path} = $newtagid2path;
				$tag_template_data_ref->{tag2_name} = $display_tag2;
			}

but I think the one you need may be the canonical id, which is in $canon_tagid and $canon_tagid2, which are not passed to the template yet. So you can add $tag_template_data_ref->{canon_tagid} = $canon_tagid;

Also don't you need to pass a language code (and possibly a country code) to facet_kp in order to get the panel in the right language?

Reply:
Yes, I have added(in my latest commit) country code to get translated data :-)